File: android/src/main/kotlin/au/com/annon/flutter_mapbox_turn_by_turn/models/MapboxProgressChangeEvent.kt
package au.com.annon.flutter_mapbox_turn_by_turn.models

import com.mapbox.navigation.base.trip.model.RouteProgress

class MapboxProgressChangeEvent(progress: RouteProgress) {
    private val isProgressChangeEvent = true
    private var currentLegDistanceRemaining: Float? = null
    private var currentLegDistanceTraveled: Float? = null
    private var currentStepInstruction: String? = null
    private var distance: Float? = null
    private var distanceTraveled: Float? = null
    private var duration: Double? = null
    private var legIndex: Int? = null

    init {
        currentLegDistanceRemaining = progress.currentLegProgress?.distanceRemaining
        currentLegDistanceTraveled = progress.currentLegProgress?.distanceTraveled
        currentStepInstruction = progress.bannerInstructions?.primary()?.text()
        distance = progress.distanceRemaining
        distanceTraveled = progress.distanceTraveled
        duration = progress.durationRemaining
        legIndex = progress.currentLegProgress?.legIndex
    }
}

File: ios/Classes/models/MapboxProgressChangeEvent.swift
import Foundation
import MapboxCoreNavigation
import MapboxDirections
import MapboxNavigation

public class MapboxProgressChangeEvent: Codable {
  let isProgressChangeEvent: Bool
  let currentLegDistanceRemaining: Double
  let currentLegDistanceTraveled: Double
  let currentStepInstruction: String
  let distance: Double
  let distanceTraveled: Double
  let duration: Double
  let legIndex: Int

  init(progress: RouteProgress) {
    isProgressChangeEvent = true
    currentLegDistanceRemaining = progress.currentLegProgress.distanceRemaining
    currentLegDistanceTraveled = progress.currentLegProgress.distanceTraveled
    currentStepInstruction = progress.currentLegProgress.currentStep.description
    distance = progress.distanceRemaining
    distanceTraveled = progress.distanceTraveled
    duration = progress.durationRemaining
    legIndex = progress.legIndex
  }
}
